jubjub/src/lib.rs

```diff
@@ -638,6 +677,36 @@ impl ExtendedPoint {
         self.to_niels().multiply(by)
     }
 
+    /// Converts a batch of projective elements into affine elements.
+    ///
+    /// This function will panic if `p.len() != q.len()`.
+    ///
+    /// This costs 5 multiplications per element, and a field inversion.
+    fn batch_normalize(p: &[Self], q: &mut [AffinePoint]) {
+        let mut acc = Fq::one();
```

Comment:
Should have an assertion that p.len() == q.len()

Reply:
The default implementation of this trait method does have the assertion, but I'd forgotten to propagate it to the overridden functions.